lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -368,15 +377,17 @@ prefilterBlock b u0 = runState $ do
         , amount = Quantity amt
         }
     applyTx
-        :: IsOurs s Address
+        :: (IsOurs s Address, IsOurs s ChimericAccount)
         => ([(Tx, TxMeta)], UTxO)
         -> Tx
         -> State s ([(Tx, TxMeta)], UTxO)
     applyTx (!txs, !u) tx = do
         ourU <- state $ utxoOurs tx
         let ourIns = Set.fromList (inputs tx) `Set.intersection` dom (u <> ourU)
         let u' = (u <> ourU) `excluding` ourIns
-        let received = fromIntegral @_ @Integer $ balance ourU
+        ourWithdrawals <- mapMaybeM ourChimericAccount $ Map.toList $ withdrawals tx
+        let received =
+                fromIntegral @_ @Integer $ (sum ourWithdrawals) + balance ourU
         let spent = fromIntegral @_ @Integer $ balance (u `restrictedBy` ourIns)
         let amt = fromIntegral $ abs (received - spent)
```

Comment:
Okay, now this is only missing the `reclaim` from the input balance when there are some :+1:

Reply:
Although... the reclaims aren't explicit in the transaction. There's no "input" for them. You know that there are some by looking at the _fees_ and the delta between input and output :thinking: ...

So, right now, if we discover a transaction with a reclaimed deposit, it may have a negative amount (because the "spent" will likely be bigger than the total "received").... This is a bit out of scope for this PR however, so may you just open a ticket about taking reclaims into account in the primitive model when prefiltering blocks?

Reply:
https://github.com/input-output-hk/cardano-wallet/issues/1944